src/components/NotificationCenter/index.js

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { values } from "lodash";
+import { connect } from "react-redux";
+import Notification from "./Notification";
+
+
+export function NotificationCenter ({notifications}) {
```

Comment:
React, dans le cas des functional components, ce qui est le cas ici, va appeler à chaque nouveau render la fonction avec comme premier paramètre les `props`. 

`props` est un objet avec des clés-valeurs. Les clés sont des strings, les valeurs peuvent être n'importe quoi.
Dans notre cas tes props sont les suivantes:
```javascript
{ 
  "notifications": Array<Notification>,
  "dismissNotification": fonction
}
```

Les clés sont donc les strings `"notifications"` et `"dismissNotification"`. Les valeurs sont `Array<Notification>` et `function`.

Dans ton cas, tu écris
`export default function NotificationCenter({ notifications }) { ... }`
Quand tu écris `({ notifications })`, il faut comprendre que tu déstructures l'objet `props` passé en paramètre (take the time to read and understand [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)) pour récupérer directement ce qu'il contient dans des variables. J'ai l'impression que c'est cette partie qui t'embrouille.

En fait, si c'est plus simple pour toi, pour le moment tu peux faire:
```jsx
export function NotificationCenter (props) {
  const { notifications, dismissNotification } = props; // A chaque fois que tu donneras des nouvelles props à NotificationCenter, il faudra que tu le récupères dans les props, comme on récupère notifications et dismissNotification dans cet exemple

  // ... reste du code
}
```

Reply:
Understand that 

This
```jsx
export function NotificationCenter (props) {
  const { notifications, dismissNotification } = props;

  // ... reste du code
}
```

And this
```jsx
export function NotificationCenter ({ notifications, dismissNotification }) {
  // ... reste du code
}
```

are equivalent.
Use the first version if it is clearer to you for now, don't bother with the second. Do what you understand, the more technical aspects will come later.

Reply:
ohlala mais I need to stop fumer la moquette prcq je me disais bien qu'il fallait mettre "dismissNotification" au même endroit que "notifications" mais du coup je sais plus comment je l'avais écrit pour que ça ne marche toujours pas mdr it is terrible................. Bref, thank you ^^